scripts/e2e-ci.sh

```diff
@@ -79,125 +82,140 @@ task() {
         step Downloading granite-7b-lab model to train
-        ilab model download --repository instructlab/granite-7b-lab
+        ilab model download --repository ${GRANITE_7B_MODEL}
     fi
-    task Downloading Models Complete
+    task Downloading models Complete
 }
 
 test_list() {
     task List the Downloaded Models
-
-    if [ "$LARGE" -eq 1 ]; then
-        # step Listing the models for the large size use case
-        ilab model list | grep ${GRANITE_7B_MODEL}
-        ilab model list | grep ${PROMETHEUS_8X7B_MODEL}
-        ilab model list | grep ${MIXTRAL_8X7B_MODEL}
-    fi
+    ilab model list
     task Model Listing Complete
 }
 
-test_serve() {
-    task Serve the model
-    local model_path
-    local serve_args
-
-    serve_args=()
-    model_path="${1:-}"
-    if [ -n "$model_path" ]; then
-        serve_args+=("--model-path" "${model_path}")
-    fi
-
-
-    # output serve log while `ilab model serve` is starting up
-    touch serve.log
-    tail -f serve.log &
-    TPID=$!
-
-    ilab model serve "${serve_args[@]}" &> serve.log &
-    wait_for_server start
-    kill "$TPID"
-    task Model Serving Complete
-}
-
-test_chat() {
-    task Chat with the model
-    CHAT_ARGS=("--endpoint-url" "http://localhost:8000/v1")
-    ilab model chat -m "${TRAINED_MODEL_PATH}" -qq "${CHAT_ARGS[@]}" 'Say "Hello" and nothing else\n'
-    task Chat with the model complete
-}
-
 test_taxonomy() {
     task Update the taxonomy
 
-    # add compositional skill to the taxomony
+    step Add compositional skill to the taxomony
     mkdir -p "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/inference/qualitative/e2e-siblings
     cp "$SCRIPTDIR"/test-data/e2e-qna-freeform-skill.yaml "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/inference/qualitative/e2e-siblings/qna.yaml
 
-    # add grounded skill to the taxomony
+    step Add grounded skill to the taxomony
     mkdir -p "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/answerability/e2e-yes_or_no
     cp "$SCRIPTDIR"/test-data/e2e-qna-grounded-skill.yaml "$DATA_HOME"/instructlab/taxonomy/compositional_skills/extraction/answerability/e2e-yes_or_no/qna.yaml
 
-    # add knowledge to the taxomony
+    step Add knowledge to the taxomony
     mkdir -p "$DATA_HOME"/instructlab/taxonomy/knowledge/phoenix/overview/e2e-phoenix
     cp "$SCRIPTDIR"/test-data/e2e-qna-knowledge-phoenix.yaml "$DATA_HOME"/instructlab/taxonomy/knowledge/phoenix/overview/e2e-phoenix/qna.yaml
 
-    step Taxonomy Verification
+    step Verify the taxonomy
     ilab taxonomy diff
-    task Taxonomy Updated and Verification complete
+
+    task Taxonomy updates and verification Complete
 }
 
 test_generate() {
+    # TODO
     task Generate synthetic data
-    ilab data generate
-    task Synthetic Data Generation Complete
+    local generate_args
+    generate_args=("--endpoint-url" "http://localhost:8000/v1" "--num-cpus" "$(nproc)" --taxonomy-path='./taxonomy') # needed?
```

Comment:
These aren't needed. Anything that's set as a flag should be set in `test_init` (and then go away with system profiles). The endpoint url is not necessary since we're starting up the teacher model in the background via the command.

The large t-shirt size shouldn't need `num_cpus` set. Maybe the medium size one does depending on the model.
The `taxonomy_path` should be set in the config in `test_init`.

Reply:
This was one of the questions I had for you and Charlie - let's talk more about it later, but what you're saying makes sense!